acceptancetests/assess_updateseries.py

```diff
@@ -0,0 +1,373 @@
+    :raises AssertionError: non juju failures in the process.
+    """
+    unit = "{}/{}".format(charm_app, unit_num)
+    try:
+        output = client.get_juju_output(
+            'ssh', unit, 'sudo do-release-upgrade -f ' +
+            'DistUpgradeViewNonInteractive', timeout=3600)
+    except subprocess.CalledProcessError as e:
+        raise AssertionError(
+            "do-release-upgrade failed on {}: {}".format(unit, e))
+
+    log.info("do-release-upgrade response: ".format(output))
+
+    if reboot in 'before':
+        reboot_unit(client, unit, False)
+
+    # NOTE: attempts to use calls such as client.juju() or remote.ssh() from
+    # this point on return timeouts.  The reboot above timesout, though a
+    # reboot does occur.  Subsequent attempts to access the unit time out.
+    # The cause is unknown, but suspected to be in the CI packages as
+    # subprocess works and is used from here on out.
+
+    assert_correct_series(client, unit, to_series)
+
+    cmd = build_ssh_cmd(
+            client, unit, 'sudo juju-updateseries --to-series ' +
+            '{} --from-series {}'.format(to_series, from_series))
+    try:
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        raise JujuAssertionError(
+            "error running juju-updateseries on {}: {}".format(unit, e))
+
+    log.info(
+            "juju-updateseries ... succeeded:\n{}".format(output))
+
+    if "successfully copied tools and relinked agent tools" not in output:
+        raise JujuAssertionError("failure in juju-updateseries")
+
+    cmd = build_ssh_cmd(client, unit,
+                        'sudo juju-updateseries --to-series ' +
+                        '{} --from-series {} --start-agents'.format(
+                            to_series, from_series))
+    try:
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        if reboot in 'before':
+            if expectedStartAgents.format(charm_app, unit_num, unit_num) \
+                    not in output:
+                raise JujuAssertionError("above output incorrect")
+        elif reboot in 'after':
+            log.info("where does the juju ERROR output go?")
+            # if 'systemd is not fully running, please reboot to start agents'\
+            #                 not in output:
+            #     raise JujuAssertionError("error running juju-updateseries "+\
+            #  "on {}: {}".format(unit, e))
+        else:
+            log.critical("test failure: reboot no longer before or after")
+            raise
+
+    log.info('sudo juju-updateseries ... --start-agents: {}'.format(output))
+
+    if reboot in 'after':
+        reboot_unit(client, unit, True)
+    else:
+        client.wait_for_started()
+    log.info("juju-updateseries .... --start-agents, reboot {} " +
+             "command, succeeded".format(reboot))
```

Comment:
The + is not needed when the string is within parens:

``python
print("This is a string"
 " that continues on the next line"
 " but is really one line")
This is a string that continues on the next line but is really one line
```

Reply:
removed